types/libhoney.d.ts

```diff
@@ -14,35 +14,35 @@ declare module 'libhoney' {
      */
     dataset: string;
     /**
-     * Sample rate of data. If set, causes us to send 1/sampleRate of events and drop the rest.
+     * (Default 1) Sample rate of data. If set, causes us to send 1/sampleRate of events and drop the rest.
      */
     sampleRate?: number;
     /**
-     * We send a batch to the API when this many outstanding events exist in our event queue.
+     * (Default 50) We send a batch to the API when this many outstanding events exist in our event queue.
      */
     batchSizeTrigger?: number;
     /**
-     * We send a batch to the API after this many milliseconds have passed.
+     * (Default 100) We send a batch to the API after this many milliseconds have passed.
```

Comment:
This is problematic. This hides the benefit of batching.
Basically, it'll send the batch after every 100ms.
So, almost always a batch contains 1 event.

I suggest to set this to around 30 secs.
We should give some proper defaults.

Reply:
@arunoda I'm not sure if the default should be that high. For the same reason we have `sampleRate` default to 1 (meaning all), the `batchTimeTrigger` should be low in case you have bad error handling and lose a whole batch of traces on a crash. A lower default means less data loss.

In other words, I think we should consider batching as opt-in. If you want to take advantage of batching, you need to set these options appropriately.

cc @hharnisc

Reply:
After chatting with @styfle - these are defaults that the underlying library use and I think we should stick with them. With a long window like 30 we increase the opportunity to let the user do the wrong thing like missing events or even OOM errors due to too many events piling up.

I'd rather the user be intentional about setting a long window.